sdk/python/kfp/v2/compiler/compiler.py

```diff
@@ -659,6 +659,9 @@ def _group_to_dag_spec(
           deployment_config.executors[importer_exec_label].importer.CopyFrom(
               subgroup.importer_spec)
 
+        if hasattr(subgroup, 'enable_caching'):
```

Comment:
If user never called `set_caching_options`, `hasattr` would return `False`.
However, we want to enable caching by default.